flink-runtime/src/main/java/org/apache/flink/runtime/concurrent/FutureUtils.java

```diff
@@ -90,6 +90,21 @@
 		}
 	}
 
+	/**
+	 * Fakes asynchronous execution by immediately executing the operation and completing the supplied future
+	 * either noramlly or exceptionally.
+	 *
+	 * @param operation to executed
+	 * @param <T> type of the result
+	 */
+	public static <T> void completeFromCallable(CompletableFuture<T> future, Callable<T> operation) {
```

Comment:
I added a similar method to `FutureUtils` called `runSync` (but it creates a new future). Would it make sense to keep the names in sync? (I have no hard feeling about renaming `runSync`).

Reply:
`runSync` should probably also use your method.

Reply:
I checked out `FutureUtils.runSyn` but I couldn't find any usages of it (now and in the original commit too).
So I think we should just remove it, but also in a separate PR.
WDYT @AHeise?

Reply:
I'd probably go by hotfix commit, doesn't deserve a separate PR.

Reply:
Wanted to avoid extra CI run :)
OK, added this commit.